api.go

```diff
@@ -33,21 +33,26 @@ type APIErrorMessage struct {
 	Error  string `json:"error"`
 }
 
-func createError(errorMsg string) []byte {
-	errorObj := APIErrorMessage{"error", errorMsg}
-	responseMsg, err := json.Marshal(&errorObj)
-
-	if err != nil {
-		log.Error("Couldn't marshal error stats: ", err)
-	}
+func apiOk(msg string) APIErrorMessage {
+	return APIErrorMessage{"ok", msg}
+}
 
-	return responseMsg
+func apiError(msg string) APIErrorMessage {
+	return APIErrorMessage{"error", msg}
 }
 
-func doJSONWrite(w http.ResponseWriter, code int, responseMessage []byte) {
+func doJSONWrite(w http.ResponseWriter, code int, obj interface{}) {
+	// TODO(mvdan): remove a couple of weeks after the api refactor,
```

Comment:
Can we merge without this comment?:)

I mean can you ensure that all dojwonwrite calls will not cause panic?